iceoryx_utils/test/stresstests/test_stress_sofi.cpp

```diff
@@ -0,0 +1,545 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/internal/concurrent/sofi.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <mutex>
+#include <queue>
+#include <stdlib.h>
+#include <thread>
+
+using namespace testing;
+
+using SoFiData = int64_t;
+constexpr SoFiData INVALID_SOFI_DATA{-1};
+
+constexpr int64_t STRESS_TIME_HOURS{0};
+constexpr int64_t STRESS_TIME_MINUTES{0};
+constexpr int64_t STRESS_TIME_SECONDS{2};
+constexpr std::chrono::milliseconds STRESS_TIME{
+    ((STRESS_TIME_HOURS * 60 + STRESS_TIME_MINUTES) * 60 + STRESS_TIME_SECONDS) * 1000};
+
+class SoFiStress : public Test
+{
+  public:
+  protected:
+    virtual void SetUp()
+    {
+    }
+
+    virtual void TearDown()
+    {
+    }
+
+
+    /// @brief Sets the CPU affinity for a thread
+    ///
+    /// @param cpu is the CPU the thread shall use
+    /// @param nativeHandle is the native handle of the c++11 std::thread
+    /// @return bool: if true, setting the affinity was successfull
+    bool setCpuAffinity(unsigned int cpu, std::thread::native_handle_type nativeHandle)
+    {
+#ifndef __QNX__
+        // Create a cpu_set_t object representing a set of CPUs. Clear it and markonly cpu as set.
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        CPU_SET(cpu, &cpuset);
+        auto retVal = pthread_setaffinity_np(nativeHandle, sizeof(cpu_set_t), &cpuset);
```

Comment:
Would it be an idea to bring this stress test to mac? I would recommend it since mac os has revealed a lot of bugs in the past. But we would not be able to use this but it looks like  it can be implemented straight forward: http://www.hybridkernel.com/2015/01/18/binding_threads_to_cores_osx.html

Reply:
not shure if it's worth the effort. There is no guarantee to run on the specified cores. The idea was to prevent the threads running on the same physical core with Hyperthreading. But since this test is highly CPU bound, the OS should put them on different cores anyway